flows/arc_alto_to_json.py

```diff
@@ -0,0 +1,158 @@
+            ["node", "typescript/lib/alto/extract-text-lines-from-alto.js", url],
+            capture_output=True,
+            text=True,
+        )
+        if result.returncode != 0:
+            raise Exception(f"Error running script for {url}: {result.stderr}")
+        return result.stdout
+    except Exception as e:
+        logger.error(f"Failed to process {url}: {str(e)}")
+        raise e
+
+
+@task()
+def extract_transcript(json_string: str):
+    # process JSON
+    parsed_json = json.loads(json_string)
+    # get the full text
+    return " ".join(item["text"] for item in parsed_json["text"])
+
+
+@task(task_run_name="insert-{s3_url}-in-database")
+def insert_schema_transcript(
+    representation_id: str,
+    s3_url: str,
+    transcript: str,
+    postgres_credentials: DatabaseCredentials,
+):
+    logger = get_run_logger()
+
+    # connect to database
+    conn = psycopg2.connect(
+        user=postgres_credentials.username,
+        password=postgres_credentials.password.get_secret_value(),
+        host=postgres_credentials.host,
+        port=postgres_credentials.port,
+        database=postgres_credentials.database,
+        # connection_factory=LoggingConnection,
+    )
+    cur = conn.cursor()
+    logger.info(
+        f"Updating transcript in 'graph.representation' for {representation_id}"
+    )
+    # insert transcript into table
+    cur.execute(
+        "UPDATE graph.representation SET schema_transcript = %s WHERE id = %s",
+        (transcript, representation_id),
+    )
+    # insert url into table
+    logger.info(
+        f"Inserting {s3_url} into 'graph.schema_transcript_url' for {representation_id}"
+    )
+    cur.execute(
+        "INSERT INTO graph.schema_transcript_url (representation_id, schema_transcript_url) VALUES (%s, %s)",
+        (representation_id, s3_url),
+    )
+    conn.commit()
+
+    # Step 5: Clean up and close the connection
+    cur.close()
+    conn.close()
+
+
+@flow(
+    name="prefect_flow_arc_alto_to_json",
+    task_runner=ConcurrentTaskRunner(),
+    on_completion=[save_last_run_config],
+)
+def arc_alto_to_json(
+    s3_endpoint: str = "http://assets-int.hetarchief.be",
+    s3_bucket_name: str = "hetarchief",
+    s3_block_name: str = "arc-object-store",
+    db_block_name: str = "local",
+    full_sync: bool = False,
+):
+
+    # Load credentials
+    postgres_creds = DatabaseCredentials.load(db_block_name)
+    s3_creds = AwsCredentials.load(s3_block_name)
+    client_parameters = AwsClientParameters(endpoint_url=s3_endpoint)
+
+    # Figure out start time
+    if not full_sync:
+        last_modified_date = get_last_run_config("%Y-%m-%d")
+
+    url_list = get_url_list.submit(
+        postgres_creds,
+        since=last_modified_date if not full_sync else None,
+    ).result()
+    for representation_id, url in url_list:
+        json_string = run_node_script.submit(url=url)
+        transcript = extract_transcript.submit(json_string=json_string.result())
```

Comment:
Calling result only here is interesting, I wonder if it keeps task results from being kept in memory until the end of the flow run.

Reply:
depends on what prefect does when `result()` is called. It does live on in whatever python var it's in, but might be garbage collected faster.